packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,759 @@
+                  height: width,
+                  child: Opacity(
+                    alwaysIncludeSemantics: true,
+                    opacity: _extendedLabelFadeValue(),
+                    child: Align(
+                      alignment: AlignmentDirectional.centerStart,
+                      child: styledLabel,
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          );
+        }
+        break;
+      case NavigationRailLabelType.selected:
+        final double appearingAnimationValue = 1 - _positionAnimation.value;
+        final double lerpedPadding = lerpDouble(_verticalDestinationPaddingNoLabel, _verticalDestinationPaddingWithLabel, appearingAnimationValue);
+        content = Container(
+          constraints: BoxConstraints(
+            minWidth: width,
+            minHeight: width,
+          ),
+          padding: const EdgeInsets.symmetric(horizontal: _horizontalDestinationPadding),
+          child: ClipRect(
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                SizedBox(height: lerpedPadding),
+                themedIcon,
+                Align(
+                  alignment: Alignment.topCenter,
+                  heightFactor: appearingAnimationValue,
+                  widthFactor: 1.0,
+                  child: Opacity(
+                    alwaysIncludeSemantics: true,
+                    opacity: selected ? _normalLabelFadeInValue() : _normalLabelFadeOutValue(),
+                    child: styledLabel,
+                  ),
+                ),
+                SizedBox(height: lerpedPadding),
+              ],
+            ),
+          ),
+        );
+        break;
+      case NavigationRailLabelType.all:
+        content = Container(
+          constraints: BoxConstraints(
+            minWidth: width,
+            minHeight: width,
+          ),
+          padding: const EdgeInsets.symmetric(horizontal: _horizontalDestinationPadding),
+          child: Column(
+            children: <Widget>[
+              const SizedBox(height: _verticalDestinationPaddingWithLabel),
+              themedIcon,
+              styledLabel,
+              const SizedBox(height: _verticalDestinationPaddingWithLabel),
+            ],
+          ),
+        );
+        break;
+    }
+
+    final ColorScheme colors = Theme.of(context).colorScheme;
+    return Semantics(
+      container: true,
+      selected: selected,
+      child: Stack(
+        children: <Widget>[
+          Material(
+            type: MaterialType.transparency,
+            clipBehavior: Clip.none,
+            child: InkResponse(
+              onTap: onTap,
+              onHover: (_) {},
+              highlightShape: BoxShape.rectangle,
+              borderRadius: BorderRadius.all(Radius.circular(width / 2.0)),
+              containedInkWell: true,
+              splashColor: colors.primary.withOpacity(0.12),
```

Comment:
Should the splash/hover color be based on the icon/text colors if those are provided?

Reply:
I think they could, but then again it might cause confusion whether it comes from the icon or the label.

Reply:
gotcha, fine as-is then.